addons/l10n_in_pos/models/account_move.py

```diff
@@ -5,10 +5,10 @@
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    @api.depends('pos_session_ids')
+    @api.depends('pos_session_ids', 'reversed_pos_order_id')
     def _compute_l10n_in_state_id(self):
         res = super()._compute_l10n_in_state_id()
-        to_compute = self.filtered(lambda m: m.country_code == 'IN' and not m.l10n_in_state_id and m.journal_id.type == 'general' and m.pos_session_ids)
+        to_compute = self.filtered(lambda m: m.country_code == 'IN' and not m.l10n_in_state_id and m.journal_id.type == 'general' and (m.pos_session_ids or m.reversed_pos_order_id))
```

Comment:
Not important for the fix in itself, but is anyone ever going to overwrite this Place of Supply with something else?  (because otherwise we could just fallback on move.company_id.state_id where we need the value.